R/system.R

```diff
@@ -1938,18 +1938,35 @@ read_excel_file <- function(path, sheet = 1, col_names = TRUE, col_types = NULL,
-    } else {
-      df <- openxlsx::read.xlsx(xlsxFile = path, sheet = sheet, colNames = col_names, startRow = skip+1, na.strings = na, skipEmptyRows = skipEmptyRows, skipEmptyCols = skipEmptyCols, check.names = check.names, detectDates = detectDates)
+    # On Windows, if the path has multibyte chars, work around error from readxl::read_excel by copying the file to temp directory.
+    if (Sys.info()[["sysname"]] == "Windows" && grepl("[^ -~]", path)) {
+      new_path <- tempfile(fileext = stringr::str_c(".", tools::file_ext(path)))
+      file.copy(path, new_path)
+      if (n_max != Inf) {
+        df <- openxlsx::read.xlsx(xlsxFile = new_path, rows=(skip+1):n_max, sheet = sheet, colNames = col_names, na.strings = na, skipEmptyRows = skipEmptyRows, skipEmptyCols = skipEmptyCols , check.names = check.names, detectDates = detectDates)
+      } else {
+        df <- openxlsx::read.xlsx(xlsxFile = new_path, sheet = sheet, colNames = col_names, startRow = skip+1, na.strings = na, skipEmptyRows = skipEmptyRows, skipEmptyCols = skipEmptyCols, check.names = check.names, detectDates = detectDates)
+      }
+      # Preserve original column name for backward comaptibility (ref: https://github.com/awalker89/openxlsx/issues/102)
+      # Calling read.xlsx again looks inefficient, but it seems this is the only solution suggested in the above issue page.
+      colnames(df) <- openxlsx::read.xlsx(xlsxFile = new_path, sheet = sheet, rows = (skip+1), check.names = FALSE, colNames = FALSE) %>% as.character()
+      file.remove(new_path)
+    }
+    else {
+      if (n_max != Inf) {
+        df <- openxlsx::read.xlsx(xlsxFile = path, rows=(skip+1):n_max, sheet = sheet, colNames = col_names, na.strings = na, skipEmptyRows = skipEmptyRows, skipEmptyCols = skipEmptyCols , check.names = check.names, detectDates = detectDates)
+      } else {
+        df <- openxlsx::read.xlsx(xlsxFile = path, sheet = sheet, colNames = col_names, startRow = skip+1, na.strings = na, skipEmptyRows = skipEmptyRows, skipEmptyCols = skipEmptyCols, check.names = check.names, detectDates = detectDates)
+      }
+      # Preserve original column name for backward comaptibility (ref: https://github.com/awalker89/openxlsx/issues/102)
+      # Calling read.xlsx again looks inefficient, but it seems this is the only solution suggested in the above issue page.
+      colnames(df) <- openxlsx::read.xlsx(xlsxFile = path, sheet = sheet, rows = (skip+1), check.names = FALSE, colNames = FALSE) %>% as.character()
     }
     # trim white space needs to be done first since it cleans column names
     if(trim_ws == TRUE) {
       # use trimws from base to remove ending and trailing white space for character columns
-      df <- data.frame(lapply(df, function(x) if(class(x)=="character") trimws(x) else(x)), stringsAsFactors=F)
+      df <- df %>% dplyr::mutate_if(is.character, funs(trimws(.)))
```

Comment:
since `mutate_if` is going to be obsolete, we might want to use ` dplyr::mutate(dplyr""across(is.character, trimws))` ?

Reply:
Thank you for pointing it out. Changed it.
Re-tested low memory mode to make sure it works.